pyxem/signals/indexation_results.py

```diff
@@ -121,7 +121,84 @@ def _get_best_match(z):
         array with shape (5,)
 
     """
-    return z[np.argmax(z[:, -1]),:] 
+    return z[np.argmax(z[:, -1]), :]
+
+
+def results_dict_to_crystal_map(results, phase_key_dict, diffraction_library=None):
+    """Export an indexation result from
+    :func:`index_dataset_with_template_rotation` to a crystal map with
+    `n_best` rotations, score, mirrors and one phase ID per data point.
+
+    Parameters
+    ----------
+    results : dict
+        Results dictionary obtained from
+        :func:`index_dataset_with_template_rotation`.
+    phase_key_dict : dict
+        Dictionary mapping phase ID to phase name, obtained from
+        :func:`index_dataset_with_template_rotation`.
+    diffraction_library : diffsims.libraries.DiffractionLibrary, optional
+        Used for the structures to be passed to
+        :class:`orix.crystal_map.PhaseList`.
+
+    Returns
+    -------
+    orix.crystal_map.CrystalMap
+        Crystal map containing `results`. The map has multiple rotations
+        and properties ("correlation", "mirrored_template",
+        "template_index") per point only if `n_best` passed to
+        :func:`index_dataset_with_template_rotation` were more than one
+        and "phase_index" only has one phase.
+
+    Notes
+    -----
+    Phase's :attr:`~orix.crystal_map.Phase.point_group` must be set
+    manually to the correct :class:`~orix.quaternion.Symmetry` after
+    the crystal map is returned.
+    """
+    nx, ny, n_best = results["phase_index"].shape
+    n_points = nx * ny
+
+    # Best matching phase
+    phase_id = results["phase_index"][:, :, 0].ravel()
+    n_phases = np.unique(phase_id).size
+
+    x, y = np.indices((nx, ny)).reshape((2, n_points))
+
+    if diffraction_library is not None:
+        structures = diffraction_library.structures
+    else:
+        structures = None
+    phase_list = PhaseList(names=phase_key_dict, structures=structures)
+
+    euler = np.deg2rad(results["orientation"].reshape((n_points, n_best, 3)))
+    if n_phases > 1:
+        euler = euler[:, 0]  # Keep best match only
+    euler = euler.squeeze()  # Remove singleton dimensions
+    rotations = Rotation.from_euler(euler, convention="bunge", direction="crystal2lab")
+
+    props = {}
+    for key in ("correlation", "mirrored_template", "template_index"):
+        try:
+            prop = results[key]
+        except KeyError:
+            warn(f"Property '{key}' was expected but not found in `results`")
```

Comment:
My bad, `prop` doesn't exist if exception is raised, should continue on next iteration then

```suggestion
        except KeyError:
            warn(f"Property '{key}' was expected but not found in `results`")
            continue
```